src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1888,9 +1906,13 @@ private enum SettingId : ushort
             MaxConcurrentStreams = 0x3,
             InitialWindowSize = 0x4,
             MaxFrameSize = 0x5,
-            MaxHeaderListSize = 0x6
+            MaxHeaderListSize = 0x6,
+            EnableConnect = 0x8
         }
 
+        internal TaskCompletionSourceWithCancellation<bool> InitialSettingsReceived { get; private set; } = new TaskCompletionSourceWithCancellation<bool>();
```

Comment:
```suggestion
        internal TaskCompletionSourceWithCancellation<bool> InitialSettingsReceived { get; private set; } = new TaskCompletionSourceWithCancellation<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Reply:
Is this only ever completed / awaited when HTTP/2 web sockets are used?  If so, should we lazily initialize it rather than allocating one for every HTTP/2 connection?

Reply:
Yes, it is awaited only when websockets are used. However, I cannot figure out how to lazily initialize it in case when we create connection with non-websocket request and then reuse it for websockets i.e. with the second stream. Initial settings are already received but are not saved in `InitialSettingsReceived` because at that point we don't know about future websocket request.

The parameter is not required, it is `TaskCreationOptions.RunContinuationsAsynchronously` by default https://github.com/dotnet/runtime/blob/32d0360b73bd77256cc9a9314a3c4280a61ea9bc/src/libraries/Common/src/System/Threading/Tasks/TaskCompletionSourceWithCancellation.cs#L13